src/vctrs.h

```diff
@@ -276,19 +276,23 @@ bool duplicated_any(SEXP names);
 /**
  * @member out A vector of size `n_row` containing the output of the
  *   row wise data frame operation.
- * @member row_known A logical vector of size `n_row`. Initially, all values
- *   are initialized to `FALSE`. As we iterate along the columns, we flip the
- *   corresponding row's `row_known` value to `TRUE` if we can determine the
+ * @member row_known A boolean array of size `n_row`. It is intended to be
```

Comment:
I'm not sure we need to mention "managed by a RAWSXP", this seems like a distraction. Instead I'd just say something like "Allocated on the R heap". Also good to keep as is.